lib/src/misc/simplify.dart

```diff
@@ -3,53 +3,66 @@
 
 import 'package:latlong2/latlong.dart';
 
-double _getSqDist(
-  LatLng p1,
-  LatLng p2,
-) {
-  final double dx = p1.longitude - p2.longitude;
-  final double dy = p1.latitude - p2.latitude;
-  return dx * dx + dy * dy;
+// Custom point due to math.Point<double> being slow. Math operations tend to
+// have 20+x penalty for virtual function overhead given the reified nature of
+// Dart generics.
+class DoublePoint {
```

Comment:
Good catch, that's a heavy penalty! Context: https://github.com/dart-lang/sdk/issues/53912

We could also:
```dart
class Point extends proj4.Point {
  Point({required super.x, required super.y});

  Point operator +(Point other) => Point(x: x + other.x, y: y + other.y);
}
```
If we want to be a bit more succinct (although it would conflict with `math.Point`, so the `math` imports would need to be named, a matter of taste).

Aligning our point type with `proj4.Point` could perhaps be convenient, but not strictly necessary (it also uses named arguments, for good or for bad).

Another option is to call it `PointDouble`, might be easier to catch with auto-completion if someone is looking for a point.

Reply:
Definitely in favour of `PointDouble`, it matches closer to what is expected with using generics.

Reply:
Also, if the PR is accepted, pre-merge, will we go over the use of `math.Point` in the repo and replace with this point implementation? Keeping two separate point implementations intended to represent the same thing seems like it would fragment the API a bit.

Reply:
> Aligning our point type with proj4.Point could perhaps be convenient, but not strictly necessary (it also uses named arguments, for good or for bad).

I'd prefer not to. If PointDouble remains internal it doesn't matter and if it becomes public we hand over ownership of a key API to another project.

> Another option is to call it PointDouble, might be easier to catch with auto-completion if someone is looking for a point.

Happy to rename don't feel very strong about names.

> Also, if the PR is accepted, pre-merge, will we go over the use of math.Point in the repo and replace with this point implementation? Keeping two separate point implementations intended to represent the same thing seems like it would fragment the API a bit.

Note that at the moment PointDouble is pretty much internal. If we were to make it public, I agree that we shouldn't have N point flavors in our APIs.

Reply:
Will mark this conversation as resolved based on you feedback. Please re-open if you feel the issue has not been addressed.